### PR TITLE
feat(cli): user management via CLI — create / list / update

### LINKED ISSUE #67: API Endpoints - Users
## Context

The backend already has a `User` model and working authentication, but it still lacks admin panel API endpoints for user management. Without this, an administrator cannot create accounts, change roles, or activate/deactivate users from within the application.

## Task checklist

- add a `/users` router accessible only to admins
- add a user listing endpoint
- add a user creation endpoint
- add an endpoint to update roles and basic user data
- add an endpoint to activate / deactivate an account
- protect self-updates from accidental admin lockout
- add integration tests for admin/viewer permissions
- add validation and error-path tests

## Dependencies

- after `#49` Auth service + router
- useful before fully closing the frontend admin panel scope

### PR DESCRIPTION
**Pull request by bihius**
## Summary

Implements #219.

- Adds `scripts/manage_users.py` — a multi-subcommand argparse CLI for operator-level user management (`create`, `list`, `update`), following the same convention as the existing `seed_admin.py`
- Closes #67 with a **CLI-only approach**: user management stays at the operator layer (smaller attack surface, no HTTP endpoint to secure/audit, avoids the bootstrap chicken-and-egg problem)
- Adds 39 integration tests covering all happy paths, error exits, and lockout-guard edge cases
- Adds `make users ARGS="..."` convenience target (mirrors `make seed`)

## What changed

| File | Description |
|------|-------------|
| `src/backend/scripts/manage_users.py` | New CLI — `create`, `list`, `update` subcommands |
| `src/backend/scripts/__init__.py` | Empty init to enable package import from pytest |
| `src/backend/tests/integration/test_manage_users.py` | 39 integration tests (39/39 pass, ruff ✓, mypy strict ✓) |
| `Makefile` | `make users ARGS="list --json"` target for in-container use |

## Design decisions

**Validation** is single-sourced through the existing `UserCreate` / `UserUpdate` Pydantic schemas — email format and 12-char password minimum are not re-implemented.

**Lockout guard** — `update` refuses to demote or deactivate a user who is the last active admin. The guard counts other active admins before committing; if zero remain, the operation is rejected with a clear error.

**Output** uses `UserResponse` — `hashed_password` is never printed to stdout or JSON output.

**No new dependencies** — stdlib `argparse` only, matching the established `seed_admin.py` convention.

## Usage

```bash
# local
uv run python scripts/manage_users.py create \
    --email alice@example.com --password changeme1234 --full-name "Alice" [--role admin]

uv run python scripts/manage_users.py list [--role admin|viewer] [--active|--inactive] [--json]

uv run python scripts/manage_users.py update alice@example.com \
    [--email …] [--full-name …] [--role …] [--password …] [--activate|--deactivate]

# Docker (or via Makefile)
make users ARGS="list --json"
make users ARGS="create --email ops@example.com --password changeme1234 --full-name Ops"
```

## Test plan

- [x] `uv run --extra dev pytest tests/integration/test_manage_users.py -v` → 39/39 pass
- [x] `ruff check scripts/manage_users.py tests/integration/test_manage_users.py` → clean
- [x] `mypy scripts/manage_users.py tests/integration/test_manage_users.py` → no issues (strict mode)

### CODE REVIEW

**Comment by bihius** [comment]
False positive addressed in c1a0d70 — the flagged string (`changeme1234`) was example text in the module docstring and argparse epilog, not a real credential. Replaced both occurrences with `'<password>'` to make the placeholder intent unambiguous.

File: src/backend/scripts/manage_users.py [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+    try:
+        validated = UserCreate(
+            email=email,
+            password=password,
+            full_name=full_name,
+            role=UserRole(role),
+        )
+    except ValidationError as exc:
+        print(f"Error: {_format_validation_error(exc)}", file=sys.stderr)
+        sys.exit(1)
```
[the remark this answers is not in the document]
Comment (bihius) [reply src/backend/scripts/manage_users.py:152]: Fixed in 6290df0 — added a `except ValueError` clause before `except ValidationError` in `cmd_create` so an invalid role string prints a clean `Error: role: ...` message and calls `sys.exit(1)`.

File: src/backend/scripts/manage_users.py [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+    role: str | None = None,
+    active_filter: bool | None = None,
+    as_json: bool = False,
+) -> None:
+    """List users with optional filters."""
+    query = db.query(User)
+    if role is not None:
+        query = query.filter(User.role == UserRole(role))
+    if active_filter is not None:
+        query = query.filter(User.is_active == active_filter)
```
[the remark this answers is not in the document]
Comment (bihius) [reply src/backend/scripts/manage_users.py:202]: Fixed in 6290df0 — wrapped `UserRole(role)` in `cmd_list` with `try/except ValueError` that prints a clean error and exits with code 1.

File: src/backend/tests/integration/test_manage_users.py [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+        assert user.role == UserRole.admin
+
+    def test_default_role_is_viewer(self, db: Session) -> None:
+        cmd_create(
+            db,
+            email="carol@test.com",
+            password="password12345",
+            full_name="Carol",
+            role="viewer",
+        )
```
[the remark this answers is not in the document]
Comment (bihius) [reply src/backend/tests/integration/test_manage_users.py]: Fixed in 6290df0 — renamed to `test_creates_viewer_user_with_explicit_role` to accurately reflect that the test verifies creation with an explicit `role="viewer"` argument, not the argparse default.
